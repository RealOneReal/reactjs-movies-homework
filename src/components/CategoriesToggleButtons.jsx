import { useState } from "react";
import { ToggleButtonGroup, ToggleButton } from "@mui/material";

const CategoriesToggleButtons = () => {
  const [categorie, setCategorie] = useState("popular");
  const handleChangeCategorie = (event) => {
    setCategorie(event.target.value);
  };

  const styles = {
    group: {
      float: "right",
      mt: 2,
      mr: 3,
    },
    button: {
      color: "secondary.light",
    },
  };

  return (
    <ToggleButtonGroup
      value={categorie}
      color="primary"
      onChange={handleChangeCategorie}
      exclusive
      sx={styles.group}
    >
      <ToggleButton value="popular" sx={styles.button}>
        Popular
      </ToggleButton>
      <ToggleButton value="top_rated" sx={styles.button}>
        Top rated
      </ToggleButton>
      <ToggleButton value="upcoming" sx={styles.button}>
        Upcoming
      </ToggleButton>
    </ToggleButtonGroup>
  );
};

export default CategoriesToggleButtons;
