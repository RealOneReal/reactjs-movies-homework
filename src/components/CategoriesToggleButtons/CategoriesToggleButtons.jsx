import { useDispatch, useSelector } from "react-redux";
import { useEffect } from "react";
import { ToggleButtonGroup, ToggleButton } from "@mui/material";
import {
  fetchGenresConfig,
  fetchMoviesByCategory,
} from "../../redux/movies/moviesThunks";
import { categoryChange } from "../../redux/search/searchSlice";
import { API_MOVIES } from "../../api/moviesAPI";

const CategoriesToggleButtons = () => {
  const dispatch = useDispatch();

  const { category } = useSelector((state) => state.search);
  const language = useSelector((state) => state.search.language);
  const inputValue = useSelector((state) => state.search.value);
  const handleChangeCategorie = ({ target }) => {
    if (inputValue) {
      return;
    }
    dispatch(categoryChange(target.value));
    dispatch(
      fetchMoviesByCategory({ category: target.value, pageNumber: 1, language })
    );
  };

  const styles = {
    group: {
      display: "flex",
      justifyContent: {
        md: "flex-end",
        xs: "center",
      },
      mr: 3,
    },
    button: {
      color: "secondary.light",
    },
  };

  useEffect(() => {
    if (inputValue) {
      return;
    }
    dispatch(fetchGenresConfig({ language }));
    dispatch(fetchMoviesByCategory({ category, pageNumber: 1, language }));
  }, [language]);

  return (
    <ToggleButtonGroup
      value={category}
      color="primary"
      onChange={handleChangeCategorie}
      exclusive
      sx={styles.group}
    >
      <ToggleButton value={API_MOVIES.POPULAR} sx={styles.button}>
        Popular
      </ToggleButton>
      <ToggleButton value={API_MOVIES.TOP_RATED} sx={styles.button}>
        Top rated
      </ToggleButton>
      <ToggleButton value={API_MOVIES.UPCOMING} sx={styles.button}>
        Upcoming
      </ToggleButton>
    </ToggleButtonGroup>
  );
};

export default CategoriesToggleButtons;
