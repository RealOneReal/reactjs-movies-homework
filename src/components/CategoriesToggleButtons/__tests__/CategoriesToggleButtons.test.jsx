import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import CategoriesToggleButtons from "../CategoriesToggleButtons";
import userEvent from "@testing-library/user-event";

describe("CategoriesToggleButtons", () => {
  it("should be render", () => {
    const tree = renderer.create(<CategoriesToggleButtons />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should have three buttons", () => {
    const { getAllByRole } = render(<CategoriesToggleButtons />);
    const buttons = getAllByRole("button");
    expect(buttons.length).toBe(3);
  });
  it("should have active button after click", () => {
    render(<CategoriesToggleButtons />);
    const upcomingButton = screen.getByText(/upcoming/i);
    userEvent.click(upcomingButton);
    const isPressed = upcomingButton.getAttribute("aria-pressed");
    expect(isPressed).toBeTruthy();
  });
});
