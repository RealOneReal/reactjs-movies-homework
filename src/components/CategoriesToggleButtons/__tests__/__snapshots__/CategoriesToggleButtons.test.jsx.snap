// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesToggleButtons should be render 1`] = `
<div
  className="MuiToggleButtonGroup-root css-cttit5-MuiToggleButtonGroup-root"
  role="group"
>
  <button
    aria-pressed={true}
    className="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1omrkh-MuiButtonBase-root-MuiToggleButton-root"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
    value="popular"
  >
    Popular
  </button>
  <button
    aria-pressed={false}
    className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1omrkh-MuiButtonBase-root-MuiToggleButton-root"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
    value="top_rated"
  >
    Top rated
  </button>
  <button
    aria-pressed={false}
    className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1omrkh-MuiButtonBase-root-MuiToggleButton-root"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
    value="upcoming"
  >
    Upcoming
  </button>
</div>
`;
